src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1324,6 +1325,59 @@ def create_or_replace_view(
             source_plan,
         )
 
+    def find_table_function_in_sql_tree(self, plan: SnowflakePlan):
+        """This function is meant to find any udtf function call from a create dynamic table plan and
+        replace '*' with explicit column identifier in the select of table function. Since we cannot
+        differentiate udtf call from other table functions, we apply this change to all table functions.
+        """
+        deepcopied_plan = copy.deepcopy(plan)
+        queue = deque()
+        queue.append(deepcopied_plan)
+        from snowflake.snowpark._internal.analyzer.select_statement import (
+            SelectTableFunction,
+            Selectable,
+        )
+
+        while queue:
+            deepcopied_plan = queue.popleft()
+            for node in deepcopied_plan.children_plan_nodes:
+                queue.append(node)
+
+            # the bug only happen when create dynamic table on top of a table function
+            # this is meant to decide whether the plan is select from a tale function
+            if isinstance(deepcopied_plan, SelectTableFunction) and isinstance(
+                deepcopied_plan.snowflake_plan.source_plan, TableFunctionJoin
+            ):
```

Comment:
Is it possible that we miss `if isinstance(deepcopied_plan, SnowflakePlan) and isinstance(deepcopied_plan.source_plan, TableFunctionJoin)`?

Reply:
- this judgement shall find all udtf inside a create dynamic table
- if there is no match, the function equals to no-op, should not affect anything